assets/js/app.js

```diff
@@ -1,21 +1,137 @@
-// Brunch automatically concatenates all files in your
-// watched paths. Those paths can be configured at
-// config.paths.watched in "brunch-config.js".
-//
-// However, those files will only be executed if
-// explicitly imported. The only exception are files
-// in vendor, which are never wrapped in imports and
-// therefore are always executed.
-
-// Import dependencies
-//
-// If you no longer want to use a dependency, remember
-// to also remove its path from "config.paths.watched".
 import "phoenix_html"
+import socket from "./socket"
+import { createPlayerNameInput, createOpponentsList, createStartGameButton } from "./view"
+import { runWithState } from "./util"
+const uuid = require('uuid/v1')
 
-// Import local files
-//
-// Local files can be imported directly using relative
-// paths "./socket" or full ones "web/static/js/socket".
+const getPlayerId = () => {
+    let playerId = document.cookie.replace(/(?:(?:^|.*;\s*)player_id\s*\=\s*([^;]*).*$)|^.*$/, "$1")
 
- import socket from "./socket"
+    if (!playerId) {
+        playerId = uuid()
+
+        setPlayerId(playerId)
+    }
+
+    return playerId
+}
+
+const setPlayerId = (player_id) => {
+    document.cookie = `player_id=${player_id}`
+}
+
+const redrawer = (channel) => (state, mapState) => {
+  const player_id = state.current.view.player_id
+  const player_name = state.current.game.player_names[player_id]
+  const game = document.getElementById("game")
+
+  while (game.hasChildNodes()) {
+    game.removeChild(game.lastChild)
+  }
+
+  const player_name_input = createPlayerNameInput(player_name, (event) => {
+    channel.push("player_update", { game_id, player_id, player_name: event.target.value })
+  })
+  const join_game_link = document.createElement('p')
+  join_game_link.appendChild(document.createTextNode(`Join Game Link: ${window.location}`))
+
+  const game_header = document.createElement('h2')
+  game_header.appendChild(document.createTextNode(`Joined Game: ${state.current.game.name}`))
+
+  const start_game_button = createStartGameButton(() => {
+    channel.push("go", { "game_id": game_id })
+  })
+
+  game.appendChild(game_header)
+  game.appendChild(join_game_link)
+  game.appendChild(player_name_input)
+  game.appendChild(createOpponentsList(player_id, state.current.game.player_names))
+  game.appendChild(start_game_button)
+  player_name_input.focus()
+}
+
+const initializer = (channel) => (mapState) => {
+  // Begin garbage code
+  let timeLeft = 0
+  let updateTimer = function () {
+    document.getElementById("timer").innerText = timeLeft
+    timeLeft -= 1
+  }
+  let timerInterval
+  // End garbage code
+
+  const player_id = getPlayerId()
+
+  // channels
+  channel.on("player_update", (game) => {
+    console.log(game)
+    mapState((state) => ({
+      current: {
+        game,
+        view: state.current.view,
+      },
+      previous: state.previous
```

Comment:
I realize this is missing the point a bit, but for clarification, would this need to be this? 
```suggestion
      previous: {
        game: state.current.game,
        view: state.previous.view
      }
```

Reply:
Yep, totally missed that. Nice catch!